Add AI agent support

jwhonce opened the pull request:
* Add AGENTS.md

inknos commented on AGENTS.md:
```diff
@@ -0,0 +1,356 @@
+
+### Coverage Requirements
+
+- Minimum 80% coverage (enforced by tox -e coverage)
```
thanks copilot we are working on it

inknos reviewed (commented):
just couple of comments on a quick scan, I'll add more in a second round

inknos reviewed (changes requested):
Went through it again and with more attention. I am ok to move on with this and merge it once the `.cursorignore` comment is addressed :)

jwhonce committed:
Add AI agent support

* Add AGENTS.md with project layout, testing workflow, code
  standards, and upstream expectations for AI coding assistants

Signed-off-by: Jhon Honce <jhonce@redhat.com>
Assisted-by: Cursor:claude-opus-4-6

Honny1 reviewed (approved):
LGTM

PTAL @inknos
